cloudinit/sources/DataSourceAzure.py

```diff
@@ -554,6 +554,43 @@ def _is_ephemeral_networking_up(self) -> bool:
             or self._ephemeral_dhcp_ctx.lease is None
         )
 
+    def _check_azure_proxy_agent_status(self) -> None:
+        """Check if azure-proxy-agent is ready for communication with WS/IMDS.
+
+        If ProvisionGuestProxyAgent is true, query azure-proxy-agent status,
+        waiting up to 120 seconds for the proxy to negotiate with Wireserver
+        and configure an eBPF proxy.  Once azure-proxy-agent is ready,
+        it will exit with code 0 and cloud-init can then expect to be able to
+        communicate with these services.
+
+        Fail deployment if azure-proxy-agent is not found or otherwise returns
+        an error.
+
+        For more information, check out:
+        https://github.com/azure/guestproxyagent
+        """
+        try:
+            cmd = [
+                "azure-proxy-agent",
+                "--status",
+                "--wait",
+                "120",
+            ]
+            out, err = subp.subp(cmd)
+            report_diagnostic_event(
+                "Running azure-proxy-agent %s resulted"
+                "in stderr output: %s with stdout: %s" % (cmd, err, out),
+                logger_func=LOG.debug,
+            )
+        except FileNotFoundError:
+            error = errors.ReportableErrorProxyAgentNotFound()
+            self._report_failure(error)
+        except subp.ProcessExecutionError as error:
+            reportable_error = errors.ReportableErrorProxyAgentStatusFailure(
+                error
+            )
+            self._report_failure(reportable_error)
```

Comment:
What I don't particularly like about this logic, is that there appears to be nothing reported currently to me on stdout or stderr when this command exits 1 on unsupported environments (lxd containers using the preliminary jammy deb package linked)  . Can we add a sample out/err string that we expect to the existing unit tests so we can see how this failure expectation looks due to lack of response from the wireserver?

Reply:
Hey, I wonder if I understand correctly. In addition to the remote logging, which could fail on unsupported environments, and the cloud init logs which would work even on unsupported environments, you'd like us to log in stderr?

I'm also not sure about the testing part. Do you want this logging to happen only for the sake of the unit test? If so, I could mock/assert the cloud init logging logic, although we have unit tests for it.

Thank you for your time!